torch/utils/_pytree.py

```diff
@@ -140,58 +146,88 @@ def _register_pytree_node(
 def _dict_flatten(d: Dict[Any, Any]) -> Tuple[List[Any], Context]:
     return list(d.values()), list(d.keys())
 
-def _dict_unflatten(values: List[Any], context: Context) -> Dict[Any, Any]:
+
+def _dict_unflatten(values: Iterable[Any], context: Context) -> Dict[Any, Any]:
     return dict(zip(context, values))
 
+
 def _list_flatten(d: List[Any]) -> Tuple[List[Any], Context]:
     return d, None
 
-def _list_unflatten(values: List[Any], context: Context) -> List[Any]:
+
+def _list_unflatten(values: Iterable[Any], context: Context) -> List[Any]:
     return list(values)
 
+
 def _tuple_flatten(d: Tuple[Any, ...]) -> Tuple[List[Any], Context]:
     return list(d), None
 
-def _tuple_unflatten(values: List[Any], context: Context) -> Tuple[Any, ...]:
+
+def _tuple_unflatten(values: Iterable[Any], context: Context) -> Tuple[Any, ...]:
     return tuple(values)
 
+
 def _namedtuple_flatten(d: NamedTuple) -> Tuple[List[Any], Context]:
     return list(d), type(d)
 
-def _namedtuple_unflatten(values: List[Any], context: Context) -> NamedTuple:
+
+def _namedtuple_unflatten(values: Iterable[Any], context: Context) -> NamedTuple:
     return cast(NamedTuple, context(*values))
 
+
 def _namedtuple_serialize(context: Context) -> DumpableContext:
     json_namedtuple = {
         "class_name": context.__name__,
         "fields": context._fields,
     }
     return json_namedtuple
 
+
 def _namedtuple_deserialize(dumpable_context: DumpableContext) -> Context:
     class_name = dumpable_context["class_name"]
     assert isinstance(class_name, str)
     context = namedtuple(class_name, dumpable_context["fields"])  # type: ignore[misc]
     return context
 
-def _odict_flatten(d: 'OrderedDict[Any, Any]') -> Tuple[List[Any], Context]:
+
+def _odict_flatten(d: "OrderedDict[Any, Any]") -> Tuple[List[Any], Context]:
```

Comment:
Nit: why does this need to be a string annotation? Just import typing.OrderedDict and use it directly.